bundles/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -775,91 +786,83 @@ public State getVirtualState(KM200CommObject object, Class<? extends Item> itemT
                         .get(object.getParent()).getValueParameter());
                 String[] servicePath = service.split("/");
                 String virtService = servicePath[servicePath.length - 1];
-                switch (virtService) {
-                    case "weekday":
-                        if (itemType.isAssignableFrom(StringItem.class)) {
-                            String val = sPService.getActiveDay();
-                            if (val == null) {
-                                return null;
-                            }
-                            state = new StringType(val);
-                        } else {
-                            logger.error("Bindingtype not supported for day service: {}", itemType.getClass());
```

Comment:
Please use `logger.warn` for problems that don't lead to system stability issues.  If the binding recovers from a situation, don't use a more severe level than `WARN`.  Other similar lines below.